### ISSUE TITLE
StaticImage only works with relative file paths

### ISSUE THREAD

**Issue by cdanny358** [gatsbyjs/gatsby#35658]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

`StaticImage`, from `gatsby-plugin-image`, only seems to work when the `src`is specified as a relative file path, such as `../../images/Filter.png`. It fails with a path like `/src/images/Filter.png`.

### Reproduction Link

https://github.com/cdanny358/static-image-test

### Steps to Reproduce

1. Create a fresh Gatsby project with `npm init gatsby` and add support for responsive images.
2. Add the lines `import { StaticImage } from 'gatsby-plugin-image'` and `import Icon from '/src/images/icon.png'` up top.
3. Add the lines `<StaticImage src='/src/images/icon.png' />` and `<img src={Icon} />` somewhere in the body.
...

### Expected Result

Both the `StaticImage` and `img` should display the Gatsby icon.

### Actual Result

Only the `img` displays, while in the console it complains about not being to find the src for the `StaticImage`. Changing the StaticImage src to `../images/icon.png` causes it to work properly.

### Environment

```shell
System:
    OS: macOS 12.3.1
    CPU: (16) x64 Intel(R) Xeon(R) W-3223 CPU @ 3.50GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - /usr/local/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 8.3.2 - /usr/local/bin/npm
  Browsers:
    Chrome: 101.0.4951.64
    Firefox: 100.0
    Safari: 15.4
  npmPackages:
    gatsby: ^4.14.0 => 4.14.0
    gatsby-plugin-image: ^2.14.1 => 2.14.1
    gatsby-plugin-sharp: ^4.14.1 => 4.14.1
    gatsby-source-filesystem: ^4.14.0 => 4.14.0
    gatsby-transformer-sharp: ^4.14.0 => 4.14.0
```

**Comment by KyleAMathews** [comment]
Yes this is by design as we in general only support relative paths. Do you have a specific need for absolute paths?

**Comment by cdanny358** [comment]
Well, it's just much simpler to type a path starting in `/src` than to calculate how deeply you're nested relative to your images. Besides, src-relative paths work fine for importing images for use with an `img` tag, as you see above. Why wouldn't StaticImage be consistent with that? I also tend to set up aliases in the gatsby-node file for all the important folders in a project, so I can access the images with something like `'images/icon.png'` from anywhere. Aliases work fine for pretty much everything *except* StaticImage.

**Comment by pieh** [comment]
I'll re-clasify this as a feature request (not bug report), due to the way it works now is intentional. I can see potential value in supporting "paths relative to root" or aliases, but not supporting them is NOT a bug.

Code wise - here's where we generate full path

https://github.com/gatsbyjs/gatsby/blob/6cac9d2f76699ab627949d921d06b03bbb499541/packages/gatsby-plugin-image/src/node-apis/image-processing.ts#L128

Where `sourceDir` is directory of file being parsed https://github.com/gatsbyjs/gatsby/blob/6cac9d2f76699ab627949d921d06b03bbb499541/packages/gatsby-plugin-image/src/node-apis/preprocess-source.ts#L67

> Aliases work fine for pretty much everything except StaticImage

Webpack doesn't take part in StaticImage handling - we only do babel transform, so aliases defined for webpack just can't work. There possibly are other ways to define aliases (not scoped to webpack) that might be potentially used to achieve this feature
